docs/source/notes/modules.rst

```diff
@@ -0,0 +1,433 @@
+.. _modules:
+
+Modules
+=======
+
+PyTorch uses modules to both build and represent neural networks. Modules are:
```

Comment:
what's the distinction you are making between building and representing?  How can you build something without a representation of it?

Reply:
The distinction we're attempting to make is that:

1. Neural networks are _built from_ modules
2. Neural networks _are_ modules

It's possible to have 1 without 2. In PyTorch, both 1 and 2 hold, resulting in arguably more flexibility / composability.

Perhaps there's a less ambiguous word than "represent" to get this idea across. Alternatively, this distinction could be left for later in the note, and this line can be revised to "PyTorch uses modules to represent neural networks".

Reply:
Update: I've changed the text to avoid making this distinction. Should be more straightforward now.